### PR TITLE
security/openconnect: add pin-sha256 to certificate hash

### LINKED ISSUE #1355: OpenConnect pin-sha256
The OpenConnect UI should support [Public-Key-Pinning](https://timtaubert.de/blog/2014/10/http-public-key-pinning-explained/) format as a hash.

Example

> To trust this server in future, perhaps add this to your command line:
>     --servercert pin-sha256:SDqgu8gcbxiE487woYrZPslpdoib7+R4Xrgsj3vn8yA=

[Forum Discussion](https://forum.opnsense.org/index.php?topic=12890.msg59793#msg59793)

### PR DESCRIPTION
**Pull request by mimugmail**
Closes #1355

### CODE REVIEW

**Comment by mimugmail** [comment]
Nice catch, fixed

File: security/openconnect/src/opnsense/mvc/app/models/OPNsense/Openconnect/General.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
         </password>
         <servercert type="TextField">
             <Required>N</Required>
-            <mask>/^[a-z0-9]{40,64}$/</mask>
+            <mask>/^([a-zA-Z0-9\/\+\=]){40,64}$/u</mask>
```
Comment (fabianfrz) [inline security/openconnect/src/opnsense/mvc/app/models/OPNsense/Openconnect/General.xml]: in theory you don't need the round bracket, the count operator also works for the square brackets.
You can also omit the A-Z or the a-z if you add an i at the end (i means no case sensitive matching).

No need to change that - it's just a question of taste.

File: security/openconnect/Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 PLUGIN_NAME=		openconnect
-PLUGIN_VERSION=		1.3.3
+PLUGIN_VERSION=		1.3.4
```
Comment (fabianfrz) [inline security/openconnect/Makefile]: I would prefer 1.4 due to a mimimal API change.

In semantic versioning the revision (3rd number means only bugfixes, second means minimal changes which may require adjusting scripts etc. while a major is maybe totally incompatible or not backward compatible).

Comment (mimugmail) [reply security/openconnect/Makefile]: Indeed, small new feature, I'll bump to 1.4.0

**Comment by fichtner** [comment]
Merged, thanks! ❤️

**Pull request merged**